source/renderer/app/components/widgets/forms/InlineEditingInput.js

```diff
@@ -76,31 +97,36 @@ export default class InlineEditingInput extends Component<Props, State> {
     {
       plugins: { vjf: vjf() },
       options: {
-        validateOnChange: true,
+        validateOnChange: this.props.validateOnChange,
         validationDebounceWait: FORM_VALIDATION_DEBOUNCE_WAIT,
       },
     }
   );
 
   submit = () => {
     this.validator.submit({
-      onSuccess: (form) => {
+      onSuccess: async (form) => {
+        this.setInputBlur();
         const { inputField } = form.values();
-        if (inputField !== this.props.inputFieldValue) {
-          this.props.onSubmit(inputField);
-          this.props.onStopEditing();
-        } else {
-          this.props.onCancelEditing();
+        const { onSubmit, errorMessage } = this.props;
+        if (inputField !== this.props.value || errorMessage) {
+          await this.setState({
```

Comment:
@daniloprates Because `this.setState` returns `void` type, using `await` is not meaningful.
Please remove `await`.